doc/CMakeLists.txt

```diff
@@ -1,18 +1,25 @@
+include(FindPythonModule) # required for find_python_module
+
 find_package(Sphinx)
 add_target_if(SPHINX_FOUND
     check-sphinx
     "Finding Sphinx"
     "Sphinx must be installed to build documentation")
 
+if (SPHINX_FOUND)
+    find_python_module(svgwrite)
+    if (NOT HAVE_SVGWRITE)
+        message(WARNING "Building documentation requires the svgwrite Python module. You can obtain it from pip like so: pip3 install svgwrite --user")
```

Comment:
I would avoid giving the explicit `--user` tag: instead just say install with pip.
Best practice is to install in a virtual env, which we shouldn't nudge users away from (let's leave it up to them to manage their packages).

We could add another `add_target_if` to define a `check-py-svgwrite` target (see `check-sphinx` above), that would then be used as a dependency of the `html` target below, like `check-sphinx`.

```cmake
find_python_module(svgwrite)
add_target_if("${SPHINX_FOUND} AND ${HAVE_SVGWRITE}"
    check-py-svgwrite
    "Finding python svgwrite"
    "The python module svgwrite needs to be installed to build documentation")
```

Something like the above could be used (you will have to test and check the syntax of the boolean expression, because I don't know enough CMake off the top of my head to be certain that it is correct).

Reply:
I tried that first, but then the error doesn't show up in the `cmake` step, only in the `make html`. What's the right stage for this error to show up?

I pushed your proposed version just now.

Reply:
Yes, that was the intention to have the error occur at `make html`, i.e. when building the docs.

**Rationale** building the library/examples/tests should work even if everything required to build the docs is not available. So we defer throwing the error until doc build time, but we use the configure time test to ensure that we produce a nice error message.